packages/@aws-cdk/aws-rds/lib/private/util.ts

```diff
@@ -78,8 +78,8 @@ export function engineDescription(engine: IEngine) {
 
 export function applyRemovalPolicy(cfnDatabase: CfnResource, removalPolicy?: RemovalPolicy): void {
   if (!removalPolicy) {
-    // the default DeletionPolicy is 'Snapshot', which is fine,
-    // but we should also make it 'Snapshot' for UpdateReplace policy
+    // Even though 'SNAPSHOT' is the default for RDS, make it explicit so cfnlint won't complain
+    cfnDatabase.cfnOptions.deletionPolicy = CfnDeletionPolicy.SNAPSHOT;
     cfnDatabase.cfnOptions.updateReplacePolicy = CfnDeletionPolicy.SNAPSHOT;
   } else {
     // just apply whatever removal policy the customer explicitly provided
```

Comment:
Gotta say, I find this current function to be pretty confusing... Wouldn't something like this be equivalent?

```ts
export function applyRemovalPolicy(cfnDatabase: CfnResource, removalPolicy: RemovalPolicy = RemovalPolicy.SNAPSHOT): void {
  cfnDatabase.applyRemovalPolicy(removalPolicy);
}
```

? And if it is, maybe it's actually better to inline this function at the call sites at this point, since it's so simple?

Reply:
Yeah, true :)